glass-easel-template-compiler/src/js_bindings.rs

```diff
@@ -41,18 +49,43 @@ impl TmplGroup {
         let path = group::path::normalize(path);
         self.group
             .add_script(&path, tmpl_str)
-            .map_err(|x| JsValue::from(format!("{}", x)))?;
-        Ok(())
+            .map_err(|x| JsValue::from(format!("{}", x)))
     }
 
     #[wasm_bindgen(js_name = "getDirectDependencies")]
     pub fn get_direct_dependencies(&self, path: &str) -> Result<js_sys::Array, JsValue> {
-        let deps = self.group.get_direct_dependencies(&path).map_err(|x| JsValue::from(format!("{}", x)))?;
-        let ret = js_sys::Array::new_with_length(deps.len() as u32);
-        for dep in deps {
-            ret.push(&JsValue::from(dep));
-        }
-        Ok(ret)
+        self.group
+            .get_direct_dependencies(&path)
+            .map(|x| convert_str_arr(&x))
+            .map_err(|x| JsValue::from(format!("{}", x)))
```

Comment:
It seems that a `js_sys::Error` should be returned instead of a string?

Reply:
In other words, it should be `From<Error>` for `js_sys::Error`

Reply:
handled by `wasm_bindgen::JsError`